bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/resource/RootCategoryResource.java

```diff
@@ -65,9 +72,25 @@ public ValueMap getValueMap() {
     @Override
     public <AdapterType> AdapterType adaptTo(Class<AdapterType> type) {
         if (type.equals(ValueMap.class)) {
-            return (AdapterType) getValueMap();
+            return type.cast(getValueMap());
+        } else if (type.equals(Node.class)) {
+            return type.cast(adaptToNode(Thread.currentThread().getStackTrace()));
         }
 
         return super.adaptTo(type);
     }
+
+    Node adaptToNode(StackTraceElement[] stackTrace) {
+        // return null when adaptTo() is called by
+        // com.day.cq.dam.core.impl.servlet.FolderThumbnailServlet.getPreviewGenerator()
```

Comment:
Well I don't understand why exactly we have the NPE issue and I don't have a better proposal to fix it, but looking for that particular stacktrace method call really looks like a hack. Yes it fixes the issue, but relying on that very implementation specific fix doesn't look very nice. I had a quick look at `FolderThumbnailServlet` and it checks that the node type is `nt:folder`, so isn't it enough to just change the type (what you already did)?

Reply:
I called it a workaround but you can call it hack too... 
Changing the type is not enough. 
I don't consider this an optimal or definitive solution either and I have some comments in the issue tracker on that.

Reply:
Another possibility could to emulate the thumbnail node structure that the DAM code expects when looking for an already existing thumbnail, for example under `/var/commerce/products/graphql/folderThumbnail/jcr:content`. We could add this in the Sling resource provider, but yes this would also be a very DAM-specific workaround.

Reply:
Until [this line](https://git.corp.adobe.com/CQ/dam/blob/master/bundles/core/src/main/java/com/day/cq/dam/core/impl/ui/preview/FolderPreviewAssetIterator.java#L80) is fixed everything we do about it is a hack.